src/query/storage/m3/cluster_namespaces_watcher_test.go

```diff
@@ -113,6 +113,11 @@ func TestClusterNamespacesWatcherRegisterListenerNamespacesAlreadySet(t *testing
 	watcher.RegisterListener(listener)
 
 	<-updateCh
+
+	// Add a short sleep to ensure we don't receive a duplicate (and undesired) call
+	time.Sleep(250 * time.Millisecond)
+
+	require.Equal(t, 1, listener.callCount)
```

Comment:
Is this necessary? Looks like if you somehow get the second call, you'll get a hung channel here right?

https://github.com/m3db/m3/pull/2793/files#diff-ec49272c47edc3b710d90f2956f4d4963c1ca5e651a29b46fc8404688a26d067R145

Reply:
It's a buffered channel so the extra notification will just sit there. We could make it an unbuffered channel and timeout after a while but I think it'll fail faster this way.